firebase-perf/src/main/java/com/google/firebase/perf/application/AppStateMonitor.java

```diff
@@ -173,8 +173,9 @@ public synchronized void onActivityStopped(Activity activity) {
       if (activityToResumedMap.isEmpty()) {
         // no more activity in foreground, app goes to background.
         stopTime = clock.getTime();
-        updateAppState(ApplicationProcessState.BACKGROUND);
         sendSessionLog(Constants.TraceNames.FOREGROUND_TRACE_NAME.toString(), resumeTime, stopTime);
+        // order is important to avoid b/204362742
+        updateAppState(ApplicationProcessState.BACKGROUND);
```

Comment:
Can you also qualify the reasoning for this ordering? For example: Complete _fs trace before triggering a change in the application state as that would lead to a new sessionID change.